pkg/validation/validation.go

```diff
@@ -73,6 +84,42 @@ func containerPorts(contour *operatorv1alpha1.Contour) error {
 	return fmt.Errorf("http and https container ports are unspecified")
 }
 
+// nodePorts validates nodeports of contour, returning an error if the nodeports
+// do not meet the API specification.
+func nodePorts(contour *operatorv1alpha1.Contour) error {
+	ports := contour.Spec.NetworkPublishing.Envoy.NodePorts
+	if ports == nil {
+		// When unspecified, defaults to 30080 for http and 30443 for https.
+		return nil
+	}
+
+	var numsFound []int32
```

Comment:
use maps here instead? similarly for https://github.com/projectcontour/contour-operator/blob/939d4daf8b0ce882a25f55a61d04e41736b8cc92/pkg/validation/validation.go#L61-L62

Reply:
@sunjayBhatia commit `37b314c` simplifies this logic. If merged, I'll use a separate PR to convert the existing logic.